airflow/docker.go

```diff
@@ -663,3 +679,46 @@ func checkServiceState(serviceState, expectedState string) bool {
 	scrubbedState := strings.Split(serviceState, " ")[0]
 	return scrubbedState == expectedState
 }
+
+func startDocker() error {
+	buf := new(bytes.Buffer)
+	err := cmdExec(DockerCmd, buf, buf, "ps")
+	if err != nil {
+		// open docker
+		fmt.Println("\nIt looks like the docker engine is not running")
+		err = cmdOpenExec(OpenCmd, buf, os.Stderr, "-a", "docker")
+		if err != nil {
+			return err
+		}
+		fmt.Println("\nIf you don't see docker desktop starting exit this command and start it manually")
+		fmt.Println("If you don't have docker desktop installed install it here: https://www.docker.com/products/docker-desktop/")
+		fmt.Println("If you are using colima or another docker alternative you will have to start the engine manually")
+		fmt.Println("\nStarting up the docker engine…")
+		// poll for docker
+		err = waitForDocker(buf)
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func waitForDocker(buf io.Writer) error {
+	timeout := time.After(time.Duration(timeoutNum) * time.Second)
+	ticker := time.NewTicker(time.Duration(tickNum) * time.Millisecond)
+	for {
+		select {
+		// Got a timeout! fail with a timeout error
+		case <-timeout:
+			return errors.New("timed out waiting for docker")
+		// Got a tick, we should check on checkSomething()
+		case <-ticker.C:
+			err := cmdPsExec(DockerCmd, buf, buf, "ps")
```

Comment:
```suggestion
                        buf.Reset()
			err := cmdPsExec(DockerCmd, buf, buf, "ps")
```
nitpick: It would be ideal to reset the buffer, since that's stored in memory, and it would keep acquiring memory that it does not require. Also because the `buf` is not used after that.

Note: this would require `buf` to be of type `bytes.Buffer` and not `io.Writer`. Should work fine if this is implemented with the suggestion above this.